Automattic/wp-calypso
# Partner Portal: Add a Terms of Service consent gate to access the portal

Context: p1616184557016100-slack-CQN0YDYM7

#### Changes proposed in this Pull Request

* Add a Terms of Service consent gate to access the Partner Portal.

#### Testing instructions

* If you do not have a partner key, please contact Infinity for one or you will be unable to view test this PR.
* Check out this PR locally, then run `yarn && yarn start-jetpack-cloud`.

Case 1: You don't have a partner or at least 1 active partner key.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to /partner-portal/partner which should show an error message like so:
![Screenshot 2021-03-23 at 18 04 45](https://user-images.githubusercontent.com/22746396/112178187-4166a000-8c02-11eb-8fd5-3daa46ecf3fb.png)

Case 2: You have an active partner key but you have not accepted the new ToS.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to /partner-portal/terms-of-service:
![Screenshot 2021-04-13 at 15 55 19](https://user-images.githubusercontent.com/22746396/114555757-b01ea280-9c70-11eb-847e-237247c126da.png)
* Accepting the terms should redirect you to key selection if you have multiple keys or to your license listing if you only have a single key.

Case 3: You have an active partner key and you have already accepted the new ToS once.
* Visit http://jetpack.cloud.localhost:3000/partner-portal
* You should be redirected to key selection if you have multiple keys or to your license listing if you only have a single key, skipping the ToS step.
